Quotient/networkaccessmanager.cpp

```diff
@@ -25,6 +25,7 @@ class {
     struct ConnectionData {
         QString accountId;
         HomeserverData hsData;
+        QByteArray accessToken;
```

Comment:
I'm looking at `HomeserverData` usages now and get to wonder whether we should add `accessToken` to it instead. This structure is meant to encapsulate all data necessary to form a network request to Matrix API (I should have documented it...), and the access token fits that bill perfectly. The `NAM` code could then be simplified a bit further.

From a paranoid standpoint, whoever calls `Connection::homeserverData()` would then get everything necessary to make an _authenticated_ request without thinking of it; but that method is protected anyway. Calling `connectionData()->accessToken()` would get exactly the same thing and is available today to exactly the same callers. And one still has to explicitly set the header using these data.

Reply:
done